draft-ietf-quic-invariants.md

```diff
@@ -165,11 +181,9 @@ version-specific semantics are marked with an X.
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+
-|0|C|X X X X X X|
+|0|X X X X X X X|
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                                                               |
-+                     [Connection ID (64)]                      +
-|                                                               |
+|                Destination Connection ID (0..144)           ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Given the dest conn ID is variable length now, can we maybe have the packet number before the connection id in the header?

Reply:
Packet number isn't an invariant. So it cannot precede any invariant field (Connection ID).

Reply:
Technically, it could, but there would have to be an invariant way of determining the offset of the Connection ID field.

Reply:
As Nick implied, I think the best order is version, connection ID(s), then packet number, regardless of what we decide on encrypting packet numbers.